### ISSUE TITLE
Support basic auth authentication

### ISSUE THREAD

**Issue by Nevon** [kafkajs/confluent-schema-registry#20]
If the schema registry is behind basic auth, there is no way to authenticate currently. This was [brought up by Mensah in the KafkaJS slack](https://kafkajs.slack.com/archives/CF6RFPF6K/p1571925289068200).

I suggest adding a new `auth` property to the constructor, which gets forwarded to mappersmith:

```
new SchemaRegistry({ host: 'http://registry:2181/', auth: { username: 'bob', password: '******' } })
```

**Comment by erikengervall** [comment]
#21 addresses this exact issue, but moving forward we should probably expose the forge configuration in the constructor, why not enable users to leverage the full capabilities of mappersmith :)

**Comment by Nevon** [comment]
There's pros and cons to doing that. It would be exposing an internal implementation detail, and lock you to using Mappersmith forever. Then again, it would be very powerful for users to be able to add whatever middleware they want.

**Comment by erikengervall** [comment]
> There's pros and cons to doing that. It would be exposing an internal implementation detail, and lock you to using Mappersmith forever. Then again, it would be very powerful for users to be able to add whatever middleware they want.

I agree it's an internal implementation detail. However, there's already somewhat mappersmith-specific fields in the interface (e.g. `clientId` and `retry`).

At this point a change of http client seems unlikely. I'd argue that the profit from exposing mappersmith capabilities outweighs the cons, which in this case would be e.g. a major version bump since a switch in http client wouldn't necessarily be backwards compatible. I much rather have this than people doing `registry.api._manifest.middleware.unshift(BasicAuth);` and expecting _that_ to be future proof 😅

**Comment by edgji** [comment]
maybe a "middleware" configuration option since that seems something that could easily be adapted to a different http client if that were to change.

**Comment by erikengervall** [comment]
> maybe a "middleware" configuration option since that seems something that could easily be adapted to a different http client if that were to change.

Yeah, this has been suggested but not yet implemented in #22.
